src/Tools/dotnet-stack/ReportCommand.cs

```diff
@@ -0,0 +1,206 @@
+
+        /// <summary>
+        /// Reports a stack trace
+        /// </summary>
+        /// <param name="ct">The cancellation token</param>
+        /// <param name="console"></param>
+        /// <param name="processId">The process to report the stack from.</param>
+        /// <param name="name">The name of process to report the stack from.</param>
+        /// <param name="output">The output path for the collected trace data.</param>
+        /// <param name="duration">The duration of to trace the target for. </param>
+        /// <returns></returns>
+        private static async Task<int> Report(CancellationToken ct, IConsole console, int processId, string name, TimeSpan duration)
+        {
+            string tempNetTraceFilename = Path.GetRandomFileName() + ".nettrace";
+            string tempEtlxFilename = "";
+
+            try
+            {
+                // Either processName or processId has to be specified.
+                if (!string.IsNullOrEmpty(name))
+                {
+                    if (processId != 0)
+                    {
+                        Console.WriteLine("Can only specify either --name or --process-id option.");
+                        return -1;
+                    }
+                    processId = CommandUtils.FindProcessIdWithName(name);
+                    if (processId < 0)
+                    {
+                        return -1;
+                    }
+                }
+
+                if (processId < 0)
+                {
+                    Console.Error.WriteLine("Process ID should not be negative.");
+                    return -1;
+                }
+                else if (processId == 0)
+                {
+                    Console.Error.WriteLine("--process-id is required");
+                    return -1;
+                }
+
+
+                var client = new DiagnosticsClient(processId);
+                var providers = new List<EventPipeProvider>()
+                {
+                    new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler", EventLevel.Informational)
+                };
+
+                // collect a *short* trace with stack samples
+                // the hidden '--duration' flag can increase the time of this trace in case 10ms
+                // is too short in a given environment, e.g., resource constrained systems
+                // N.B. - This trace INCLUDES rundown.  For sufficiently large applications, it may take non-trivial time to collect
+                //        the symbol data in rundown.
+                using (EventPipeSession session = client.StartEventPipeSession(providers))
+                using (FileStream fs = File.OpenWrite(tempNetTraceFilename))
+                {
+                    Task copyTask = session.EventStream.CopyToAsync(fs);
+                    await Task.Delay(duration);
+                    session.Stop();
```

Comment:
Since rundown can take a while, do we wanna put in some kind of visual cue that something is happening? I've seen some other languages CLI tools that spins ( '\', '-', '/' '-') to visually indicate progress is being made. Maybe this is something we can add here (and also to dotnet-trace if rundown is requested).

Reply:
If we need to add something I'd recommend not making it an in-place update to keep the output simple and scripting friendly. "report" is intended to be our non-interactive output verb.

Reply:
CLI tools will sometimes go to lengths to determine if stdout is redirected and provide different output for delayed consumption (remove syntax coloring, no overwrite of previous drawn text...), but I think keeping it simple is easier.

Reply:
Ok - at the very least I think it would be worthwhile to add a note here, if it's taking a long time. We've had numerous issues come in both on GitHub and internally from first party customers asking why dotnet-trace is "stuck".

Reply:
I expect `report` to be scriptable in much the same way the `perf script` verb is.  Specifically, the output needs to be consistent in shape.  Showing a non-data message should be an all or nothing decision since inconsistency makes it more difficult to script around the output.

I'd prefer if users were unaware that this does a trace under the covers.  We're intending to build a more direct, simplified mechanism for getting a single stack sample in the future, so I see the trace and rundown steps as implementation details unrelated to the functionality.

My preference is to not have a message, but I could add one.  Would the following be acceptable?

```
$ dotnet-stack report -n stacktest31
# Sufficiently large applications may cause this command to take non-trivial amounts of time
Thread (0xBA4FCB):
  [Native Frames]
  System.Console!System.IO.StdInReader.ReadKey(bool&)
  System.Console!System.IO.StdInReader.ReadLine(bool)
  System.Console!System.IO.StdInReader.ReadLine()
  System.Console!System.IO.SyncTextReader.ReadLine()
  System.Console!System.Console.ReadLine()
  stacktest31!stacktest31.Program.Main(class System.String[])
```

If we preface all non-stack text with a known comment delimiter, e.g., `#`, `%`, or `//`, it should be easy enough to script around.

Reply:
I added a message that prints after 5 seconds that is prefixed with a `#` to easily ignore it if scripting.